### PR TITLE
fix(ci): tests workflow now downloads artifacts

### PR DESCRIPTION
**Pull request by mbrsamsung**

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -29,10 +29,14 @@ jobs:
...
+    - name: Fetch python build artifact
+      uses: actions/download-artifact@v2
+      with:
+        name: python-build-artifacts
+
     - name: build and install
       run: |
-        pip install build
-        make clean build install
+        make install
```
Comment (safl) [inline .github/workflows/tests.yml]: How does it know to use downloaded artifacts for the install?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -7,20 +7,23 @@ on:
...
+    - '*'
 
 defaults:
   run:
     shell: bash
 
 jobs:
   build:
-    name: Build
+    name: Build python package
```
Comment (safl) [inline .github/workflows/build.yml:20]: This commit also does:

* fix typos
* bump upload and checkout actions
* add trigger on pull-request
* expand 'build release package'

aka there is a lot going on there, perhaps just state that in the commit-message

**Review by safl — COMMENTED** [review]
One concern about the the downloaded artifact is actually used, otherwise just some misc. comments.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (mbrsamsung) [reply .github/workflows/tests.yml]: `make install` always uses the build artifact to install spex

File: .github/workflows/build.yml (hunk 2 above)
Comment (mbrsamsung) [reply .github/workflows/build.yml:20]: I have expanded the git message

**Review by safl — APPROVED** [review]
LGTM!

**Pull request merged**